File: June_week2/괄호의값.cpp
#include <iostream>
#include <vector>
using namespace std;

int main() {
    string data;
    vector<int> v;
    cin >> data;
    for (int i = 0; i < data.length(); i++) {
        if (data[i] == '(' || data[i] == '[')
            v.push_back(data[i]);
        else if (data[i] == ')' || data[i] == ']') {
            if (v.back() == '(' || v.back() == '[') {
                int num = v.back() == '(' ? 2 : 3;
                v.pop_back();
                if (v.back() != '(' && v.back() != '[') {
                    num *= v.back();
                    v.pop_back();
                }
                v.push_back(num);

            } else {
                int num = v.back() * data[i] == ')' ? 2 : 3;
                v.pop_back();  // 숫자 pop
                v.pop_back();  // '(' 또는 '[' 괄호 pop
                v.push_back(num);
            }
        }
        for (auto& elem : v) {
            cout << elem << ' ';
        }
        cout << endl;
    }
    cout << v[0];
}

File: June_week2/괄호의값.py
data, stack, answer = list(input()), [], 0
for elem in data:
    if elem == '(' or elem == '[':
        stack.append(elem)
    elif elem == ')' or elem == ']':
        tmp, cur_num, stopped = 0, (2 if elem == ')' else 3), False
        for s in reversed(stack):
            stack.pop()
            if (elem == ')' and s == '(') or (elem == ']' and s == '['):
                stopped = True
                break
            elif type(s) == int:
                tmp += int(s)
        if stopped:
            stack.append(cur_num if tmp == 0 else cur_num*tmp)
        else:
            print(0)
            exit(0)
if '[' in stack or '(' in stack:
    print(0)
else:
    print(sum(stack))
